hack/conformance/check_conformance_test_requirements.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This tool is for checking conformance e2e tests follow the requirements
+// which is https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+const (
+	//e.g. framework.ConformanceIt("should provide secure master service ", func() {
+	patternStartConformance = "framework.ConformanceIt\\(.*, func\\(\\) {$"
+	patternEndConformance   = "}\\)$"
+	patternSkipProviderIs   = "Skip.*ProviderIs\\("
+)
+
+var exitCode = 0
+
+// This function checks the requirement: it works for all providers (e.g., no SkipIfProviderIs/SkipUnlessProviderIs calls)
+func checkAllProviders(e2eFile string) error {
+	inConformanceCode := false
+
+	regStartConformance := regexp.MustCompile(patternStartConformance)
+	regEndConformance := regexp.MustCompile(patternEndConformance)
+	regSkipProviderIs := regexp.MustCompile(patternSkipProviderIs)
+
+	fileInput, err := ioutil.ReadFile(e2eFile)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to read file %s: %v\n", e2eFile, err)
+		return err
+	}
+	scanner := bufio.NewScanner(bytes.NewReader(fileInput))
+	scanner.Split(bufio.ScanLines)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if regStartConformance.MatchString(line) {
+			if inConformanceCode {
+				return errors.Errorf("Missed the end of previous conformance test. There might be a bug in this script.")
+			}
+			inConformanceCode = true
+		}
+		if inConformanceCode {
+			if regSkipProviderIs.MatchString(line) {
+				// To list all invalid places in a single operation of this tool, here doesn't return error and continues checking.
+				fmt.Fprintf(os.Stderr, "%v: Conformance test should not call SkipIfProviderIs()/SkipUnlessProviderIs()\n", e2eFile)
+				exitCode = 1
+			}
+			if regEndConformance.MatchString(line) {
+				inConformanceCode = false
+			}
+		}
+	}
+	if inConformanceCode {
+		return errors.Errorf("Missed the end of previous conformance test. There might be a bug in this script.")
+	}
+	return nil
+}
+
+func processFile(e2ePath string) error {
+	regGoFile := regexp.MustCompile(".*.go")
+
+	files, err := ioutil.ReadDir(e2ePath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to read dir %s: %v\n", e2ePath, err)
```

Comment:
same comment about wrapping the error and printing on the caller.

Reply:
done